vortex-cxx/CMakeLists.txt

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+cmake_minimum_required(VERSION 3.16)
+
+if(DEFINED ENV{VCPKG_TOOLCHAIN_PATH})
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_TOOLCHAIN_PATH}")
+else()
+    message(WARNING "VCPKG_TOOLCHAIN_PATH environment variable is not set. Dependencies may not be found correctly. See README.md for vcpkg setup instructions.")
+endif()
+
+project(vortex-cxx LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE Debug)
+endif()
+
+option(VORTEX_ENABLE_TESTING "Enable building test binary for vortex-cxx" OFF)
```

Comment:
We should enable `-Wall -Wpedantic` for the CXX build.

Reply:
Added with an extra `-Wno-dollar-in-identifier-extension` to prevent warnings from cxx generated code.